challenger/src/grinding_challenger.rs

```diff
@@ -0,0 +1,37 @@
+use p3_field::PrimeField64;
+use p3_symmetric::CryptographicPermutation;
+
+use crate::{DuplexChallenger, FieldChallenger};
+
+pub trait GrindingChallenger<F: PrimeField64>: FieldChallenger<F> + Clone {
+    // Can be overridden to allow more efficient ways to fork, not cloning the entire state.
+    fn fork(&self) -> Self {
```

Comment:
Hm would the expected behavior different from `Clone`? If not, I think we might as well use `Clone`, since an optimization that works for one should work for the other.

I'm guessing the goal is to support optimizations along the lines of Plonky2's, but I think we may need to override `grind` for that. Like in general, if we can't make assumptions about how the forked challenger may be used, it seems like it may be necessary to copy input or output buffers; Plonky2's grinding code was sort of getting around that by understanding how the forked challenger was going to be used.

Reply:
Hmm yeah that makes sense!